physionet-django/project/models.py

```diff
@@ -858,18 +824,104 @@ def citation_text(self):
 
             elif style == 'Vancouver':
 
+                style_format = ('{author}. {title} (version {version}). '
+                                'PhysioNet. {year}')
+
+                doi_format = ('. Available from: '
+                              '<a href="https://doi.org/{doi}">'
+                              'https://doi.org/{doi}</a>.')
+
                 all_authors = ', '.join(a.initialed_name()
-                                         .replace(',', '')
-                                         .replace('.', '') for a in authors)
+                                        .replace(',', '')
+                                        .replace('.', '') for a in authors)
 
-            citation_format = format_html(style_list[count],
-                                          author=all_authors,
-                                          **shared_content)
+            if doi:
+                final_style = style_format + doi_format
+                citation_format = format_html(final_style,
+                                              author=all_authors,
+                                              doi=doi,
+                                              **shared_content)
+
+            else:
+                final_style = style_format + '.'
+                citation_format = format_html(final_style,
+                                              author=all_authors,
+                                              **shared_content)
 
             citation_dict[style] = citation_format
 
         return citation_dict
 
+
+class SubmissionInfo(models.Model):
```

Comment:
Not clear why, but the diff suggests that these classes have been deleted and then re-added (when I think they've just been shifted down). Were they reformatted in some way? It's best to avoid this happening, because it makes it difficult for us to see what has changed.

Reply:
I'm not sure why it did that either... It's possible it was reformatted but I thought I just added the `citation_text()` function just above them.. Will look into it further!

Reply:
Notice how if you go to "Files Changed" then it doesn't show up? It must've been something funky about the commit but I think it is good now.